scout/server/extensions/loqus_extension.py

```diff
@@ -48,86 +49,220 @@ class LoqusDB:
+            instance_settings(dict) : The settings of a specific loqusdb instance
+
+        Returns:
+            version (float)
+        """
+        version = None
+        if instance_settings["instance_type"] == "api":
+            version = self.get_api_loqus_version(api_url=instance_settings.get(API_URL))
+        else:
+            version = self.get_exec_loqus_version(loqusdb_id=instance_settings.get("id"))
+
+        if not version >= 2.5:
             LOG.info("Please update your loqusdb version to >=2.5")
             raise EnvironmentError("Only compatible with loqusdb version >= 2.5")
+        return version
 
     @staticmethod
-    def set_coordinates(variant_info):
-        """Update coordinates depending on variant type
+    def get_api_loqus_version(api_url):
+        """Get version of LoqusDB instance available from a REST API"""
+        if api_url is None:
+            return None
+        json_resp = api_get("".join([api_url, "/"]))
+        return float(json_resp.get("loqusdb_version"))
+
+    def get_exec_loqus_version(self, loqusdb_id=None):
+        """Get LoqusDB version as float"""
 
-        Some SVs have tricky ways to represent coordinates.
+        call_str = self.get_command(loqusdb_id)
+        call_str.extend(["--version"])
+        LOG.debug("call_str: {}".format(call_str))
+        try:
+            output = execute_command(call_str)
+        except CalledProcessError:
+            LOG.warning("Something went wrong with loqus")
+            return -1.0
 
-        - Insertions have the same start and end position but they have a length
-        - Some of them have a unknown length which will then be -1.
+        version = output.rstrip().split(" ")[-1]
+        LOG.debug("version: {}".format(version))
+        return float(version)
 
-        This function will calculate the end position based on these assumptions.
+    def search_loqus_instance(self, id):
+        """Search settings for a LoqusDB instance by id
+
+        Returns:
+            {'binary_path':(str), 'id': (str), 'config_path': (str), 'instance_type': 'exec'} or
+            {'api_url':(str), 'instance_type':'api', 'id': (str)}
+        """
+        for i in self.loqusdb_settings:
+            if i.get("id") == id:
+                return i
+        return None
+
+    def case_count(self, variant_category, loqusdb_id="default"):
+        """Returns number of cases in loqus instance
+
+        Args:
+            loqusdb_id(string)
+
+        Returns:
+            nr_cases(int)
+        """
+        nr_cases = 0
+        loqus_instance = self.search_loqus_instance(loqusdb_id)
+        if loqus_instance is None:
+            LOG.error(f"Could not find a Loqus instance with id:{loqusdb_id}")
+            return nr_cases
+
+        # loqus queried via executable
+        if loqus_instance.get("instance_type") == "exec":
+            case_call = self.get_command(loqusdb_id)
+            case_call.extend(["cases", "--count", "-t", variant_category])
```

Comment:
I've added the last 2 params to this program call because I think the number of cases should be counted by type of variants. Like if you are watching a SNV then the total number of cases should be n_cases with SNVs. It is not like this on master.

Reply:
They would typically be similar for run of the mill RD wgs, but considering e.g. SVs for cancer or for exomes that makes sense.